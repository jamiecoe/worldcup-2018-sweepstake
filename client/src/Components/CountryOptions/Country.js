import React from 'react';
import styled, { css } from 'styled-components';

const StyledDiv = styled.div`
  margin: 10px;
  text-align: center;
  position: relative;  
  min-width: 240px;
  min-height: 125px;

  ${props =>
    props.knockedOut &&
    css`
      &:before {
        content: '';
        display: block;
        position: absolute;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
        background: rgba(0, 0, 0, 0.5);
      }
    `};
`;

const OverlayDiv = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
  width: 100%;
  opacity: 0;
  transition: 0.4s ease;
  background-color: rgba(0, 0, 0, 0.6);
  display: flex;
  justify-content: center;
  align-items: center;

  &:hover {
    opacity: 0.9;
  }
`;

const StyledImg = styled.img`
  box-shadow: 0 0 4px 1px rgba(0, 0, 0, 0.3);
  height: 100%;  
  width: 100%;
`;

const Country = props => {
  return (
    <StyledDiv knockedOut={props.knockedOut}>
      <StyledImg src={props.imgUrl} alt={`${props.name} flag`} />
      <OverlayDiv>
        <h3>{props.name}</h3>
      </OverlayDiv>
    </StyledDiv>
  );
};

export default Country;
