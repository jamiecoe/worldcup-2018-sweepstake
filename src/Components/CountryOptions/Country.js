import React from 'react';

const Country = (props) => {
  console.log(props);
  
  return (
    <div>
      <img src={props.imgUrl} alt={`${props.name} flag`}/>
      <h3>{props.name}</h3>
    </div>
  )
}

export default Country;
