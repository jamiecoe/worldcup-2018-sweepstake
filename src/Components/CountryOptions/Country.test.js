import React from 'react';
import { shallow } from 'enzyme';
import Country from './Country';

describe('Country', () => {
  const props = { name: 'Test Country', imageUrl: 'https://test.com/test-image.jpg' };
  const country = shallow(<Country {...props}/>);

  it('renders properly', () => {
    expect(country).toMatchSnapshot();
  });
});