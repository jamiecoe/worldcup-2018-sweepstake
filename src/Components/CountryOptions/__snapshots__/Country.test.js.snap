// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Country
    imageUrl="https://test.com/test-image.jpg"
    name="Test Country"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Test Country flag"
          src={undefined}
/>,
        <h3>
          Test Country
</h3>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Test Country flag",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Country",
        },
        "ref": null,
        "rendered": "Test Country",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Test Country flag"
            src={undefined}
/>,
          <h3>
            Test Country
</h3>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Test Country flag",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test Country",
          },
          "ref": null,
          "rendered": "Test Country",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
